### PR TITLE
Add judicial administration from Cayman Islands

### PR DESCRIPTION
**Pull request by aivuk**
The crawler gets all current Cayman Islands judges from  https://www.judicial.ky/judicial-administration/judges.

  1. The types of courts are extracted from the page menu.
  2. The courts types are used to generate the ocupancies. If a judge has the description "(Assigned to all Divisions)", all courts extracted on 1 will be assigned to the judge. If a judge has a description "(Assigned to all Divisions except the Financial Services Division)", all courts except Financial Services will be assigned to it. Some judges have the court information written on the text block. Two judges don't have the information written about a specific court.

### CODE REVIEW

**Comment by aivuk** [comment]
@jbothma @pudo Two judges (Raj Parker and David Charles Doyle) don't have any information about their specific courts. What kind of position do you suggest to create,  "Judge from Cayman Islands Judicial Administration"?

**Comment by jbothma** [comment]
I think just "Judge" is totally fine

File: datasets/ky/judicial/crawler.py [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+                name="Judge for {}".format(position),
+                country="Cayman Islands",
+            )
+            occupancy = h.make_occupancy(
+                context,
+                judge_proxy,
+                pos,
+                True,
+                start_date="2021",
+                end_date="2025",
```
Comment (jbothma) [inline datasets/ky/judicial/crawler.py:84]: I don't think we can make this assumption here.

The more I look at these, the more I think the detail is hurting us more than it's helping us. We can put a ton of effort into parsing dates from the court positions, only for them to be considered too old, yet they're listed under Current.

So I think simply emit a position "Judge" and occupancy without start or end date for each of the people under Current, making sure you don't include the Retired ones by accident.

File: datasets/ky/judicial/crawler.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+                new_judge["positions"] = []
+                for c in courts:
+                    if c in line:
+                        new_judge["positions"] += [c]
+
+        judges += [new_judge]
+
+        judge_proxy = context.make("Person")
+        h.apply_name(judge_proxy, full=new_judge["name"])
+        # TODO: Make a better ID
```
Comment (jbothma) [inline datasets/ky/judicial/crawler.py:68]: TODone :)

**Comment by jbothma** [comment]
Sorry to change scope under your feet. But could you add these
Screenshot_2023-09-04_16-24-20

with the positions

- Chief Justice
- President - Court of Appeal
- Justice of Appeal
- Judge

and include the relevant sourceUrl on each Person?

File: datasets/ky/judicial/crawler.py (hunk 2 above)
Comment (aivuk) [reply datasets/ky/judicial/crawler.py]: done!

File: datasets/ky/judicial/crawler.py (hunk 1 above)
Comment (aivuk) [reply datasets/ky/judicial/crawler.py]: done!

**Comment by aivuk** [comment]
> Sorry to change scope under your feet. But could you add these Screenshot_2023-09-04_16-24-20
> 
> with the positions
> 
>     * Chief Justice
> 
>     * President - Court of Appeal
> 
>     * Justice of Appeal
> 
>     * Judge
> 
> 
> and include the relevant sourceUrl on each Person?

done, added the sourceUrl. You can notice that part of the code can be generalized in a function, at least Chief of Justice and President of Court, I will try to improve it later, but as it is, it's getting these other positions. One question that I have is that the urls for these other persons are hardcoded, they don't come from the yml. Is there a way to add them there that you be a valid metadata?

**Comment by jbothma** [comment]
A couple of stray entries here:

**Comment by jbothma** [comment]
We can probably significantly simplify the code if we stop trying to get all this detail and just capture them all as Judge or whatever the section title maps to

```
Richard N. Williams
  ky-ja-cb07d0045b81d092042d795caa557d3507267310 ['Judge for Admiralty Division']
    status=['current'] from=[] until=[]
  ky-ja-9bd1309bd279782da1d32e279077d1bf73c367d1 ['Judge for Grand Court']
    status=['current'] from=[] until=[]
  ky-ja-b9235bc4742bd0a9058090e29e559a13ef8cc8ed ['Judge for Privy Council']
    status=['current'] from=[] until=[]
  ky-ja-1b6cbb03c0da00dacfe94676c790f457af1c83a4 ['Judge for Treatment Courts']
    status=['current'] from=[] until=[]
  ky-ja-fb87edf9c2eca0585d2701179c8b91fa9768e0f1 ['Judge for Family Division']
    status=['current'] from=[] until=[]
  ky-ja-bd19c52a5a0c165fae42c8c17f68a538366ba3ec ['Judge for Court of Appeal']
    status=['current'] from=[] until=[]
  ky-ja-52947e9e36d9e5b15cf4ed1f67335cd0231ccee9 ['Judge for Summary Court']
    status=['current'] from=[] until=[]
  ky-ja-73e76dde3de6d976752a31cbf2f7276323a921a2 ['Judge for Coroners Court']
    status=['current'] from=[] until=[]

```

**Comment by jbothma** [comment]
This one is Former, not Current, so should be excluded

```
Elliot D. Mottley KC
  ky-ja-8370629e141bd9453153e7b9a38c1c1ed9ead5d9 ['Justice of Appeal']
    status=['current'] from=[] until=[]
```

**Comment by jbothma** [comment]
Ian Kawaley is missing

**Comment by jbothma** [comment]
Let's drop all the attempt at getting detail about a person.
You can even drop the stripping of titles - at least don't put any more effort in.
Just capture the name, and the position as per the category. Judge, Justice of Appeal, etc.
Watch out the ones to exclude on Judges is "Retired" but on Justices of appeal it's "Former"

**Comment by aivuk** [comment]
Hi @jbothma sorry by the errors, most of them were happening because of a wrong xpath in the Appeal of Justice code. I also did what you suggested, removed the multiple positions for the judges, now instead of multiple positions for each court type, they just have "judge". I also simplified the code and now most of the logic is being done by crawl_judges and for each different judge type (chief justice, president of court, justices of appeal, judges) I pass different functions to get the name or the list of judges.

**Comment by aivuk** [comment]
I checked also the generated entities manually now comparing with the sources and now we have the correct number (15 in total). I also checked individually the names and positions.

File: datasets/ky/judicial/ky_judicial.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+name: ky_judicial
+title: "judicial"
```
Comment (jbothma) [inline datasets/ky/judicial/ky_judicial.yml:2]:
```suggestion
title: "Senior judicial officers of the Cayman Islands"
```

**Pull request merged**